### ISSUE TITLE
[Runtime Environment] Remove cached python libs, working dir etc

### ISSUE THREAD

**Issue by sisilmehta733** [ray-project/ray#47488]
### What happened + What you expected to happen

Currently there is no way to 
1. Remove existing cached files from the runtime environment OR
2. Launch a ray job without using any of the caching of the python libs, working dir etc.

### Versions / Dependencies

We'r using ray 2.35.0

### Reproduction script

NA

### Issue Severity

Medium: It is a significant difficulty but I can work around it.

**Comment by jjyao** [comment]
Runtime env cached files are reference counted so they will be automatically deleted when no one is using it.

> Launch a ray job without using any of the caching of the python libs, working dir etc.

If your Ray job doesn't have runtime env, then it won't use cached python libs and working dir from other jobs.

**Comment by sisilmehta733** [comment]
1. The behavior we'r hoping for is to keep the runtime environment (since we want to install dependencies). However we don't want to use the caching (some of our dependencies are internal libraries and without versioning etc).
2. We also noticed that if you pass the requirements file as "pip": ["-r requirements.txt"] this file is cached even when the contents of the file change -> which is a bug.

Any reason why caching is the default behavior and can't be changed?

**Comment by alinealfa** [comment]
@jjyao 
+1 having the above issue as described by @sisilmehta733

**Comment by nilsec** [comment]
+1

**Comment by jjyao** [comment]
cc @jiwq

**Comment by kfswain** [comment]
Agreed, would love a way to ignore the cache, as it ignores my file changes when I submit the job again.

**Comment by edvintb** [comment]
Also having this same problem.
